### PR TITLE
5934 use new clinics and providers caches

### PR DESCRIPTION
**Pull request by cadibemma**
## Summary

- The VAOS team recently added a caches for clinics and providers and methods for fetching through the cache called `get_clinic_with_cache` and `get_provider_with_cache`. We should change the proxy to use that method, since it should provide some performance benefits.

## Related issue(s)
- department-of-veterans-affairs/va-mobile-app#5934

## Testing done
- Specs

## What areas of the site does it impact?
- None

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/mobile/app/services/mobile/v2/appointments/proxy.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require 'lighthouse/facilities/client'
+require 'mobile/v0/vaos_appointments/appointments_helper'
```
Comment (kpethtel) [inline modules/mobile/app/services/mobile/v2/appointments/proxy.rb:4]: This require should not be necessary because it's within the rails app directory.

**Review by kpethtel — CHANGES_REQUESTED** [review]
Aside from that `require`, I think this is ready to go. Thanks for cleaning up the specs while you were there. I still think the stubbing in those specs is somewhat problematic because it's caused them to be very difficult to work in (as we both experienced), so I appreciate you at least removing the mocks where possible.

**Review by Tonksthebear — DISMISSED** [review]
LGTM with Kris's change

**Review by Tonksthebear — APPROVED** [review]
LGTM

**Review by kpethtel — APPROVED** [review]
Thanks for going the extra mile.

**Pull request merged**